crypto/keys/ethereum/pubkey.go

```diff
@@ -0,0 +1,85 @@
+package ethereum
+
+import (
+	"bytes"
+	"crypto/ecdsa"
+	"crypto/sha256"
+
+	etherum_crypto "github.com/ethereum/go-ethereum/crypto"
+
+	cryptotypes "github.com/cosmos/cosmos-sdk/crypto/types"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+
+	tmcrypto "github.com/tendermint/tendermint/crypto"
+)
+
+type ethereumPK struct {
+	public ecdsa.PublicKey
+}
+
+func (pk *PubKey) Address() tmcrypto.Address {
+	return tmcrypto.AddressHash(pk.Bytes())
+}
+
+func (pk *PubKey) AccAddress() sdk.AccAddress {
+	return sdk.AccAddress(pk.Address())
+}
+
+func (pk *PubKey) VerifySignature(data []byte, signature []byte) bool {
+	hashed := sha256.Sum256(data)
+
+	// Get the public key from the signature
+	sigPublicKey, err := etherum_crypto.Ecrecover(hashed[:], signature)
+	if err != nil {
+		return false
+	}
+
+	// Check if the public key recovered from the signature matches the owner
+	if !bytes.Equal(sigPublicKey, pk.Bytes()) {
+		return false
+	}
+	return true
+}
+
+func (pk *PubKey) Bytes() []byte {
+	return etherum_crypto.FromECDSAPub(&pk.Key.public)
+}
+
+func (pk *PubKey) Equals(other cryptotypes.PubKey) bool {
+	return pk.Type() == other.Type() && bytes.Equal(pk.Bytes(), other.Bytes())
+}
+
+func (pk *PubKey) Type() string {
+	return "ethereum"
+}
+
+func (pk *ethereumPK) Bytes() []byte {
+	return etherum_crypto.FromECDSAPub(&pk.public)
+}
+
+func (pk *ethereumPK) Size() int {
+	return len(pk.Bytes())
+}
+
+func (pk *ethereumPK) MarshalTo(data []byte) (int, error) {
```

Comment:
Most probably you should remove `*`

Reply:
According to https://github.com/gogo/protobuf/blob/master/custom_types.md the `MarshalTo` function should be implemented for `*`.
But after changing the format to binary topic outdated.